File: Java/tp.java
import java.util.*;

// Class Elektronik dengan 4 atribut
class Elektronik {
    private String nama;
    private String merk;
    private double harga;
    private int stok;
    
    // Constructor
    public Elektronik(String nama, String merk, double harga, int stok) {
        this.nama = nama;
        this.merk = merk;
        this.harga = harga;
        this.stok = stok;
    }
    
    // Getter methods
    public String getNama() {
        return nama;
    }
    
    public String getMerk() {
        return merk;
    }
    
    public double getHarga() {
        return harga;
    }
    
    public int getStok() {
        return stok;
    }
    
    // Setter methods
    public void setNama(String nama) {
        this.nama = nama;
    }
    
    public void setMerk(String merk) {
        this.merk = merk;
    }
    
    public void setHarga(double harga) {
        this.harga = harga;
    }
    
    public void setStok(int stok) {
        this.stok = stok;
    }
    
    // Method untuk menampilkan informasi elektronik
    public void tampilkanInfo() {
        System.out.printf("%-15s %-15s Rp %,.0f %10d%n", 
                         nama, merk, harga, stok);
    }
}

// Class utama untuk mengelola toko elektronik
public class TokoElektronik {
    private static ArrayList<Elektronik> daftarElektronik = new ArrayList<>();
    private static Scanner scanner = new Scanner(System.in);
    
    public static void main(String[] args) {
        // Menambahkan beberapa data awal
        tambahDataAwal();
        
        int pilihan;
        do {
            tampilkanMenu();
            System.out.print("Pilih menu (1-5): ");
            pilihan = scanner.nextInt();
            scanner.nextLine(); // consume newline
            
            switch (pilihan) {
                case 1:
                    tambahElektronik();
                    break;
                case 2:
                    tampilkanSemuaElektronik();
                    break;
                case 3:
                    editElektronik();
                    break;
                case 4:
                    hapusElektronik();
                    break;
                case 5:
                    System.out.println("Terima kasih telah menggunakan program Toko Elektronik!");
                    break;
                default:
                    System.out.println("Pilihan tidak valid!");
            }
            
            if (pilihan != 5) {
                System.out.println("\nTekan Enter untuk melanjutkan...");
                scanner.nextLine();
            }
        } while (pilihan != 5);
    }
    
    private static void tampilkanMenu() {
        System.out.println("\n" + "=".repeat(50));
        System.out.println("           SISTEM TOKO ELEKTRONIK");
        System.out.println("=".repeat(50));
        System.out.println("1. Tambah Elektronik");
        System.out.println("2. Tampilkan Semua Elektronik");
        System.out.println("3. Edit Elektronik");
        System.out.println("4. Hapus Elektronik");
        System.out.println("5. Keluar");
        System.out.println("=".repeat(50));
    }
    
    private static void tambahDataAwal() {
        daftarElektronik.add(new Elektronik("Laptop Gaming", "ASUS", 15000000, 5));
        daftarElektronik.add(new Elektronik("Smartphone", "Samsung", 8000000, 10));
        daftarElektronik.add(new Elektronik("Smart TV", "LG", 12000000, 3));
        daftarElektronik.add(new Elektronik("Headphone", "Sony", 2500000, 15));
    }
    
    private static void tambahElektronik() {
        System.out.println("\n--- TAMBAH ELEKTRONIK BARU ---");
        
        System.out.print("Nama Elektronik: ");
        String nama = scanner.nextLine();
        
        System.out.print("Merk: ");
        String merk = scanner.nextLine();
        
        System.out.print("Harga: ");
        double harga = scanner.nextDouble();
        
        System.out.print("Stok: ");
        int stok = scanner.nextInt();
        
        Elektronik elektronikBaru = new Elektronik(nama, merk, harga, stok);
        daftarElektronik.add(elektronikBaru);
        
        System.out.println("Elektronik berhasil ditambahkan!");
    }
    
    private static void tampilkanSemuaElektronik() {
        System.out.println("\n--- DAFTAR SEMUA ELEKTRONIK ---");
        
        if (daftarElektronik.isEmpty()) {
            System.out.println("Tidak ada data elektronik.");
            return;
        }
        
        System.out.println("No  Nama            Merk            Harga          Stok");
        System.out.println("-".repeat(65));
        
        for (int i = 0; i < daftarElektronik.size(); i++) {
            System.out.printf("%-4d", (i + 1));
            daftarElektronik.get(i).tampilkanInfo();
        }
        
        System.out.println("-".repeat(65));
        System.out.println("Total item: " + daftarElektronik.size());
    }
    
    private static void editElektronik() {
        tampilkanSemuaElektronik();
        
        if (daftarElektronik.isEmpty()) {
            return;
        }
        
        System.out.print("\nPilih nomor elektronik yang akan diedit (1-" + 
                        daftarElektronik.size() + "): ");
        int nomor = scanner.nextInt();
        scanner.nextLine();
        
        if (nomor < 1 || nomor > daftarElektronik.size()) {
            System.out.println("Nomor tidak valid!");
            return;
        }
        
        Elektronik elektronik = daftarElektronik.get(nomor - 1);
        
        System.out.println("\n--- EDIT ELEKTRONIK ---");
        System.out.println("Data saat ini:");
        System.out.printf("1. Nama: %s%n", elektronik.getNama());
        System.out.printf("2. Merk: %s%n", elektronik.getMerk());
        System.out.printf("3. Harga: Rp %,.0f%n", elektronik.getHarga());
        System.out.printf("4. Stok: %d%n", elektronik.getStok());
        
        System.out.print("\nMasukkan nama baru (Enter untuk tidak mengubah): ");
        String nama = scanner.nextLine();
        if (!nama.trim().isEmpty()) {
            elektronik.setNama(nama);
        }
        
        System.out.print("Masukkan merk baru (Enter untuk tidak mengubah): ");
        String merk = scanner.nextLine();
        if (!merk.trim().isEmpty()) {
            elektronik.setMerk(merk);
        }
        
        System.out.print("Masukkan harga baru (0 untuk tidak mengubah): ");
        double harga = scanner.nextDouble();
        if (harga > 0) {
            elektronik.setHarga(harga);
        }
        
        System.out.print("Masukkan stok baru (-1 untuk tidak mengubah): ");
        int stok = scanner.nextInt();
        if (stok >= 0) {
            elektronik.setStok(stok);
        }
        
        System.out.println("Data elektronik berhasil diupdate!");
    }
    
    private static void hapusElektronik() {
        tampilkanSemuaElektronik();
        
        if (daftarElektronik.isEmpty()) {
            return;
        }
        
        System.out.print("\nPilih nomor elektronik yang akan dihapus (1-" + 
                        daftarElektronik.size() + "): ");
        int nomor = scanner.nextInt();
        
        if (nomor < 1 || nomor > daftarElektronik.size()) {
            System.out.println("Nomor tidak valid!");
            return;
        }
        
        Elektronik elektronik = daftarElektronik.get(nomor - 1);
        System.out.printf("Yakin ingin menghapus '%s %s'? (y/n): ", 
                         elektronik.getNama(), elektronik.getMerk());
        scanner.nextLine(); // consume newline
        String konfirmasi = scanner.nextLine();
        
        if (konfirmasi.toLowerCase().equals("y")) {
            daftarElektronik.remove(nomor - 1);
            System.out.println("Elektronik berhasil dihapus!");
        } else {
            System.out.println("Penghapusan dibatalkan.");
        }
    }
}

File: C++/tp.cpp
#include <iostream>
#include <vector>
#include <string>
#include <iomanip>
#include <limits>

using namespace std;

// Class Elektronik dengan 4 atribut
class Elektronik {
private:
    string nama;
    string merk;
    double harga;
    int stok;

public:
    // Constructor
    Elektronik(string nama, string merk, double harga, int stok) {
        this->nama = nama;
        this->merk = merk;
        this->harga = harga;
        this->stok = stok;
    }
    
    // Getter methods
    string getNama() const {
        return nama;
    }
    
    string getMerk() const {
        return merk;
    }
    
    double getHarga() const {
        return harga;
    }
    
    int getStok() const {
        return stok;
    }
    
    // Setter methods
    void setNama(string nama) {
        this->nama = nama;
    }
    
    void setMerk(string merk) {
        this->merk = merk;
    }
    
    void setHarga(double harga) {
        this->harga = harga;
    }
    
    void setStok(int stok) {
        this->stok = stok;
    }
    
    // Method untuk menampilkan informasi elektronik
    void tampilkanInfo() const {
        cout << left << setw(15) << nama 
             << setw(15) << merk 
             << "Rp " << right << setw(10) << fixed << setprecision(0) << harga
             << setw(8) << stok << endl;
    }
};

// Class untuk mengelola toko elektronik
class TokoElektronik {
private:
    vector<Elektronik> daftarElektronik;
    
public:
    // Constructor
    TokoElektronik() {
        tambahDataAwal();
    }
    
    void tambahDataAwal() {
        daftarElektronik.push_back(Elektronik("Laptop Gaming", "ASUS", 15000000, 5));
        daftarElektronik.push_back(Elektronik("Smartphone", "Samsung", 8000000, 10));
        daftarElektronik.push_back(Elektronik("Smart TV", "LG", 12000000, 3));
        daftarElektronik.push_back(Elektronik("Headphone", "Sony", 2500000, 15));
    }
    
    void tampilkanMenu() {
        cout << "\n" << string(50, '=') << endl;
        cout << "           SISTEM TOKO ELEKTRONIK" << endl;
        cout << string(50, '=') << endl;
        cout << "1. Tambah Elektronik" << endl;
        cout << "2. Tampilkan Semua Elektronik" << endl;
        cout << "3. Edit Elektronik" << endl;
        cout << "4. Hapus Elektronik" << endl;
        cout << "5. Keluar" << endl;
        cout << string(50, '=') << endl;
    }
    
    void tambahElektronik() {
        cout << "\n--- TAMBAH ELEKTRONIK BARU ---" << endl;
        
        cin.ignore(); // Clear input buffer
        
        string nama, merk;
        double harga;
        int stok;
        
        cout << "Nama Elektronik: ";
        getline(cin, nama);
        
        cout << "Merk: ";
        getline(cin, merk);
        
        cout << "Harga: ";
        while (!(cin >> harga) || harga < 0) {
            cout << "Masukkan harga yang valid (positif): ";
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        }
        
        cout << "Stok: ";
        while (!(cin >> stok) || stok < 0) {
            cout << "Masukkan stok yang valid (tidak negatif): ";
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        }
        
        Elektronik elektronikBaru(nama, merk, harga, stok);
        daftarElektronik.push_back(elektronikBaru);
        
        cout << "Elektronik berhasil ditambahkan!" << endl;
    }
    
    bool tampilkanSemuaElektronik() {
        cout << "\n--- DAFTAR SEMUA ELEKTRONIK ---" << endl;
        
        if (daftarElektronik.empty()) {
            cout << "Tidak ada data elektronik." << endl;
            return false;
        }
        
        cout << "No  " << left << setw(15) << "Nama" 
             << setw(15) << "Merk" 
             << setw(15) << "Harga"
             << "Stok" << endl;
        cout << string(65, '-') << endl;
        
        for (size_t i = 0; i < daftarElektronik.size(); i++) {
            cout << left << setw(4) << (i + 1);
            daftarElektronik[i].tampilkanInfo();
        }
        
        cout << string(65, '-') << endl;
        cout << "Total item: " << daftarElektronik.size() << endl;
        return true;
    }
    
    void editElektronik() {
        if (!tampilkanSemuaElektronik()) {
            return;
        }
        
        int nomor;
        cout << "\nPilih nomor elektronik yang akan diedit (1-" 
             << daftarElektronik.size() << "): ";
        
        while (!(cin >> nomor) || nomor < 1 || nomor > (int)daftarElektronik.size()) {
            cout << "Nomor tidak valid! Masukkan nomor 1-" 
                 << daftarElektronik.size() << ": ";
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        }
        
        cin.ignore(); // Clear input buffer
        
        Elektronik& elektronik = daftarElektronik[nomor - 1];
        
        cout << "\n--- EDIT ELEKTRONIK ---" << endl;
        cout << "Data saat ini:" << endl;
        cout << "1. Nama: " << elektronik.getNama() << endl;
        cout << "2. Merk: " << elektronik.getMerk() << endl;
        cout << "3. Harga: Rp " << fixed << setprecision(0) << elektronik.getHarga() << endl;
        cout << "4. Stok: " << elektronik.getStok() << endl;
        
        string input;
        cout << "\nMasukkan nama baru (Enter untuk tidak mengubah): ";
        getline(cin, input);
        if (!input.empty()) {
            elektronik.setNama(input);
        }
        
        cout << "Masukkan merk baru (Enter untuk tidak mengubah): ";
        getline(cin, input);
        if (!input.empty()) {
            elektronik.setMerk(input);
        }
        
        cout << "Masukkan harga baru (0 untuk tidak mengubah): ";
        double harga;
        if (cin >> harga && harga > 0) {
            elektronik.setHarga(harga);
        }
        
        cout << "Masukkan stok baru (-1 untuk tidak mengubah): ";
        int stok;
        if (cin >> stok && stok >= 0) {
            elektronik.setStok(stok);
        }
        
        cout << "Data elektronik berhasil diupdate!" << endl;
    }
    
    void hapusElektronik() {
        if (!tampilkanSemuaElektronik()) {
            return;
        }
        
        int nomor;
        cout << "\nPilih nomor elektronik yang akan dihapus (1-" 
             << daftarElektronik.size() << "): ";
        
        while (!(cin >> nomor) || nomor < 1 || nomor > (int)daftarElektronik.size()) {
            cout << "Nomor tidak valid! Masukkan nomor 1-" 
                 << daftarElektronik.size() << ": ";
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        }
        
        cin.ignore(); // Clear input buffer
        
        Elektronik& elektronik = daftarElektronik[nomor - 1];
        cout << "Yakin ingin menghapus '" << elektronik.getNama() 
             << " " << elektronik.getMerk() << "'? (y/n): ";
        
        string konfirmasi;
        getline(cin, konfirmasi);
        
        if (konfirmasi == "y" || konfirmasi == "Y") {
            daftarElektronik.erase(daftarElektronik.begin() + nomor - 1);
            cout << "Elektronik berhasil dihapus!" << endl;
        } else {
            cout << "Penghapusan dibatalkan." << endl;
        }
    }
    
    void jalankan() {
        int pilihan;
        
        do {
            tampilkanMenu();
            cout << "Pilih menu (1-5): ";
            
            while (!(cin >> pilihan) || pilihan < 1 || pilihan > 5) {
                cout << "Pilihan tidak valid! Masukkan angka 1-5: ";
                cin.clear();
                cin.ignore(numeric_limits<streamsize>::max(), '\n');
            }
            
            switch (pilihan) {
                case 1:
                    tambahElektronik();
                    break;
                case 2:
                    tampilkanSemuaElektronik();
                    break;
                case 3:
                    editElektronik();
                    break;
                case 4:
                    hapusElektronik();
                    break;
                case 5:
                    cout << "Terima kasih telah menggunakan program Toko Elektronik!" << endl;
                    break;
            }
            
            if (pilihan != 5) {
                cout << "\nTekan Enter untuk melanjutkan...";
                cin.ignore();
                cin.get();
            }
        } while (pilihan != 5);
    }
};

// Program utama
int main() {
    TokoElektronik toko;
    toko.jalankan();
    
    return 0;
}